target/linux/ath79/dts/ar9344_mikrotik_routerboard-2011uias-2hnd.dts

```diff
@@ -0,0 +1,317 @@
+	};
+
+	sfp {
+		/*
+		 * SFP port attached throug AR8327 port 6 isn't functional for now
+		 * The module detects sucessfully, but lacks some GPIO definitions,
+		 * and proper SGMII configuration isn't known yet.
+		 */
+		compatible = "sff,sfp";
+		i2c-bus = <&sfp_i2c>;
+		los-gpios = <&gpio 17 GPIO_ACTIVE_HIGH>;
+		maximum-power-milliwatt = <1000>;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led-0 {
+			label = "green:wlan2g";
+			gpios = <&gpio 4 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+	};
+};
+
+&ref {
+	clock-frequency = <25000000>;
+};
+
+&gpio {
+	nand-nce {
+		gpio-hog;
+		gpios = <14 GPIO_ACTIVE_LOW>;
+		output-high;
+		line-name = "nand-nce";
+	};
+};
+
+&nand {
+	status = "okay";
+
+	nand-ecc-mode = "soft";
+	nand-ecc-step-size = <2048>;
+	qca,nand-swap-dma;
+	qca,nand-scan-fixup;
+
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		partition@0 {
+			label = "booter";
+			reg = <0x0000000 0x0040000>;
+			read-only;
+		};
+
+		partition@40000 {
+			label = "kernel";
+			reg = <0x0040000 0x03c0000>;
+		};
+
+		partition@400000 {
+			label = "ubi";
+			reg = <0x0400000 0x0>;
+		};
+	};
+};
+
+&pinmux {
+	pmx_sfp_i2c: pinmux_sfp_i2c {
+		pinctrl-single,bits = <0x8 0x00000000 0xff000000>; /* set GPIO11 back as GPIO */
+	};
+};
+
+&spi {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+
+		partitions {
+			compatible = "mikrotik,routerboot-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "routerboot";
+				reg = <0x0 0x0>;
+				read-only;
+			};
+
+			hard_config {
+				read-only;
+
+				nvmem-layout {
+					compatible = "fixed-layout";
+					#address-cells = <1>;
+					#size-cells = <1>;
+
+					macaddr_hardconfig_10: macaddr@10 {
+						compatible = "mac-base";
+						reg = <0x10 0x6>;
+						#nvmem-cell-cells = <1>;
+					};
+				};
+			};
+
+			bios {
+				size = <0x1000>;
+				read-only;
+			};
+
+			soft_config {
+			};
+		};
+	};
+};
+
+&mdio0 {
+	status = "okay";
+	phy0: ethernet-phy@0 {
+		reg = <0>;
+		phy-mode = "rgmii";
+
+		qca,ar8327-initvals = <
+			0x04 0x07c00000 /* PORT0 PAD MODE CTRL: Mac0_rgmii_en | Mac0_rgmii_txclk_delay_en | Mac0_rgmii_rxclk_delay_en | Mac0_rgmii_txclk_delay_sel3 | Mac0_rgmii_rxclk_delay_sel0 */
+			0x0c 0x00000080 /* PORT6 PAD MODE CTRL: Mac6_sgmii_en */
+			0x10 0x81000000 /* PWS_REG: POWER_ON_STRAP | LED_OPEN_EN_CSR */
+			0x50 0xc731c731 /* LED_CTRL0 */
+			0x54 0x00000000 /* LED_CTRL1 */
+			0x58 0x00008000 /* LED_CTRL2: 5A: disable LCD power */
+			0x5c 0x0230c300 /* LED_CTRL3: 5C: enable SFP power */
+			0x7c 0x0000007e /* PORT0_STATUS: DUPLEX_MODE_0 | RX_FLOW_EN_0 | TX_FLOW_EN_0 | RXMAC_EN_0 | TXMAC_EN_0 | SPEED_0 1000M */
+			0x94 0x0000007e /* PORT6_STATUS: DUPLEX_MODE_0 | RX_FLOW_EN_0 | TX_FLOW_EN_0 | RXMAC_EN_0 | TXMAC_EN_0 | SPEED_0 1000M */
+			0xe0 0xc70167d0 /* SGMII_CTRL */
```

Comment:
Could you try 0xc701e7d0 here? This is the only difference in ar8327-initvals I could spot. For me sfp autonegotiation was working fine in #3729 with a small patch you already integrated: https://github.com/openwrt/openwrt/commit/cda1a3dab0c923192b9d73fe05d56a4bf30c2c81.

Reply:
Didn't help. But it's also well possible, that all SFP-RJ45 modules I own are only gigabit-capable.

Reply:
Actually, I've noticed, that the switch can detect link status on this port after this - but something is bugged in the driver and it reports the same status for port 0 (always on, to the CPU) and port 6 (the SFP) - and that reporting happens only in kernel log, `swconfig dev switch0 show` always shows link up.